src/neomaril_codex/pipeline.py

```diff
@@ -408,17 +426,51 @@ def start(self):
             raise PipelineError("Cannot start pipeline without configuration")
 
         if self.train_config:
-            training_id = self.run_training()
+            self.__training_id = self.run_training()
         else:
-            training_id = None
+            self.__training_id = None
 
         if self.deploy_config:
-            model_id = self.run_deploy(training_id=training_id)
+            self.__model_id = self.run_deploy(training_id=self.__training_id)
         else:
-            model_id = None
+            self.__model_id = None
```

Comment:
These variables are being created outside __init__.If the variables will be manipulated, maybe define in the __init__ would be better